src/workerd/api/node/zlib-util.h

```diff
@@ -250,38 +291,113 @@ class ZlibUtil final: public jsg::Object {
         kj::Array<kj::byte> output,
         int outputOffset,
         int outputLength);
-    void writeSync(jsg::Lock& js,
-        int flush,
-        jsg::Optional<kj::Array<kj::byte>> input,
-        int inputOffset,
-        int inputLength,
-        kj::Array<kj::byte> output,
-        int outputOffset,
-        int outputLength);
-    void params(jsg::Lock& js, int level, int strategy);
     void reset(jsg::Lock& js);
 
+    JSG_RESOURCE_TYPE(CompressionStream) {
+      JSG_METHOD(close);
+      JSG_METHOD_NAMED(write, template write<true>);
+      JSG_METHOD_NAMED(writeSync, template write<false>);
+      JSG_METHOD(reset);
+      JSG_METHOD(setErrorHandler);
+    }
+
+  protected:
+    CompressionContext* context() {
```

Comment:
Our typical style would be to return a `CompressionContext&` here instead... and if you do want/need to provide the pointer, provide a separate `CompressionContext* get()` or something.